src/scheduler/server_info.c

```diff
@@ -3403,7 +3410,10 @@ read_formula(void)
 	FILE *fp;
 
 
-	sprintf(pathbuf, "%s/%s", pbs_conf.pbs_home_path, FORMULA_ATTR_PATH_SCHED);
+	if (dflt_sched)
+		sprintf(pathbuf, "%s/%s", pbs_conf.pbs_home_path, FORMULA_ATTR_PATH_SCHED);
+	else
+		sprintf(pathbuf, "%s/sched_priv_%s/%s", pbs_conf.pbs_home_path, sc_name, FORMULA_FILENAME);
```

Comment:
This is incorrect.  There is a global that holds the sched_priv name.  A scheduler's sched_priv can be called anything.  This is just the default.

Reply:
That's what my understanding was as well, but I couldn't find it. Since pbs_sched.c also does this (https://github.com/PBSPro/pbspro/blob/a280b6b9573d626db3cf055bf4f81cf6ebbcd506/src/scheduler/pbs_sched.c#L1075), I thought this would be correct. Do you now where we keep the sched_priv global? if this is incorrect then we should fix pbs_sched.c as well.

Reply:
OK, after looking through the code, I have figured out what is going on (and it is wrong).  A scheduler comes up and assumes the default sched_priv and reads the sched_config.  On the first scheduling cycle, it will read the real sched_priv from the server and does a chdir() into it.  It will not reread its sched_config though.  This means it has either read the wrong sched_config, or none at all.  It isn't until the first HUP that the real sched_config will be read.

None of this fixes your problem.  Your problem might be simpler to fix though.  Can't you just read a relative path to the sched formula file since you will be chdir()'d into the sched_priv?

Reply:
That's true, we'd already be in the sched_priv by that point, I'll make that change thanks.